src/index.cpp

```diff
@@ -142,6 +135,12 @@ Index<T, TagT, LabelT>::Index(Metric m, const size_t dim, const size_t max_point
                 (size_t)((index_parameters == nullptr ? 0 : index_parameters->max_degree) *
                          defaults::GRAPH_SLACK_FACTOR * 1.05)))
 {
+    if (_pq_dist) {
```

Comment:
Same here, if the factory knows `_pq_dist = true` won't it already initialize this with correct type ? If not shouldn't we use reset here as we have assigned value to it (line 76 in index.cpp).

Reply:
this is now the only place for pq_dist, for backward compatibility reasons